### PR TITLE
Add a macos specific tweak to .bazelrc to fix the OSX build

### LINKED ISSUE #405: OSX build broken
Following a question on the slack channel, I tried to build Nighthawk's artifacts on OSX.

The following build error occurred:

```
ERROR: /Users/oschaaf/code/envoy-perf-vscode/nighthawk/source/common/BUILD:64:1: C++ compilation of rule '//source/common:nighthawk_common_lib' failed (Exit 1) wrapped_clang failed: error executing command external/local_config_cc/wrapped_clang '-D_FORTIFY_SOURCE=1' -fstack-protector -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -fno-omit-frame-pointer -O0 -DDEBUG '-std=c++11' -iquote . -iquote ... (remaining 1191 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
In file included from source/common/statistic_impl.cc:1:
In file included from bazel-out/darwin-fastbuild/bin/source/common/_virtual_includes/nighthawk_common_lib/common/statistic_impl.h:6:
In file included from bazel-out/darwin-fastbuild/bin/include/nighthawk/common/_virtual_includes/base_includes/nighthawk/common/statistic.h:12:
In file included from bazel-out/darwin-fastbuild/bin/api/client/output.pb.h:38:
bazel-out/darwin-fastbuild/bin/api/client/options.pb.h:833:37: error: expected member name or ';' after declaration specifiers
  static constexpr VerbosityOptions DEBUG =
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
<command line>:2:15: note: expanded from here
#define DEBUG 1
              ^
1 error generated.
```

One of the proto api's has an [enum field named `DEBUG`](https://github.com/envoyproxy/nighthawk/blob/06a6a027adeb4970c853ad2f646acf28027e3fb6/api/client/options.proto#L44). On a hunch, I renamed that field, and the allows the build to succeed.
But that probably isn't an option because it would break our backwards compatibility guarantee, and I'm also not sure this is the right fix.

### PR DESCRIPTION
**Pull request by oschaaf**
Fixes #405

Signed-off-by: Otto van der Schaaf <oschaaf@we-amp.com>

### CODE REVIEW

**Comment by oschaaf** [comment]
Trying a retest here for the purpose of figuring out what's up with repokitteh:

/retest

File: ci/do_ci.sh [hunk 1]
```diff
@@ -187,9 +187,11 @@ if [ -n "$CIRCLECI" ]; then
     # As of the new coverage methodology introduced in Envoy, that has grown memory requirements too.
     # Hence we heavily reduce parallellism, to avoid being OOM killed.
     if [[ "$1" == "coverage" ]]; then
-        NUM_CPUS=4
+        NUM_CPUS=3
```
Comment (mum4k) [inline ci/do_ci.sh:190]: Can we note the motivation for these changes in the PR description?

Not sure if they are related or unintentionally added to this PR.

Comment (oschaaf) [reply ci/do_ci.sh]: Sorry - this ended up here by accident.
I backed this out of here and moved part of it into https://github.com/envoyproxy/nighthawk/pull/415.
(For coverage, I want to hold off making this change: I think an upstream Envoy change may
have lowered the memory requirements).

**Pull request merged**